ui/src/services/utils.js

```diff
@@ -372,9 +372,14 @@ export const compareHealth = (status1, status2) => {
 // Add a space between size value and its unit since the API returns this as a string
 // Add the unit B
 export const formatSizeForDisplay = (value) => {
-  if (value && value.match(/^(\d+)(\D+)$/))
-    return value.replace(/^(\d+)(\D+)$/, '$1 $2') + 'B';
-  else return value;
+  if (value && value.match(/^(\d+)(\D+)$/)) {
+    const newValue = value.replace(/^(\d+)(\D+)$/, '$1 $2');
+    if (newValue.match(/.+(B$)/i)) {
+      return newValue;
+    } else {
+      return newValue + 'B';
+    }
+  } else return value;
```

Comment:
It could fix the issue, but if it was me, I would get further to see why I have already had the unit B in the raw value. 

And I found the data `storageCapacity` is from `NodeVolumeUtile.js`

```
storageCapacity: volumeCapacityCurrent?.value[1]
    ? bytesToSize(volumeCapacityCurrent?.value[1])
    : formatSizeForDisplay(volumePV?.spec?.capacity?.storage),
```

So I would remove formatSizeForDisplay. and I think with this fix we don't need to have the regex checking.
```
storageCapacity: volumeCapacityCurrent?.value[1]
    ? bytesToSize(volumeCapacityCurrent?.value[1])
    : volumePV?.spec?.capacity?.storage,
```

Reply:
:thinking:  

Since the storage capacity is used in other components like the overviewTab. I would rather remove formatSizeForDisplay from here : https://github.com/scality/metalk8s/blob/development/2.11/ui/src/components/VolumeListTable.js#L420 . WDYT ? 

I agree that this fix doesn't solve the root issue, but since the fix was needed this week. I was more comfortable doing that because  i was sure it doesn't have side effect on other components.

Reply:
Right! OK. Let's remove the formatSizeForDisplay in the VolumeListTable then.